service/src/main/java/bio/terra/workspace/service/resource/referenced/ReferencedBigQueryDatasetResource.java

```diff
@@ -106,6 +109,12 @@ public void validate() {
     ValidationUtils.validateBqDatasetName(getDatasetName());
   }
 
+  @Override
+  public boolean validateReference(FlightBeanBag context, AuthenticatedUserRequest userReq) {
```

Comment:
nit: I think of functions starting with `validate` as void functions that throw on invalid. I wonder if there's a better name scheme for these boolean-valued ones.

Reply:
Yeah - at one point that's what this function did, but I thought it might be more useful to have an output. Renamed to `checkReferenceAccess` / `checkAccess`.